src/puppeteerActionInterceptor.ts

```diff
@@ -1,61 +1,66 @@
-function intercept(page: any, func: (actionObj: object) => void) {
-    page.on("console", async (msg: any) => {
+import type { JSHandle, Page } from "puppeteer";
+import { v4 as uuidv4 } from "uuid";
+function intercept(page: Page, func: (actionObj: object) => void) {
+    page.on("console", async (msg) => {
         if (msg.type() === "debug") {
             const text: string = msg.text();
             if (text.includes("message arrived at preview storybook/actions/action-event")) {
-                func(await unwrapJSHandle(msg.args()[2]));
+                const dataObject = await unwrapJSHandle(msg.args()[2]) as object;
+                func(dataObject);
             }
         }
     });
 }
 
-function unwrapJSHandle(handle: any) {
-    return handle.evaluate((x: any) => x);
+async function unwrapJSHandle<T>(handle: JSHandle<T>): Promise<T> {
+    return JSON.parse(await handle.evaluate((x: T) => JSON.stringify(x))) as Promise<T>;
```

Comment:
Wouldn't `parse` after `stringify` just have the same result as doing neither?

Reply:
It does indeed have the same result, except when something goes wrong.

When passing through `evaluate()`, objects are serialized with a custom protocol, not JSON. Failures in that serialization (for example, due to circular references) results in `undefined` being returned instead of a proper error being thrown, which makes the issue very hard to debug.

Doing a `JSON.stringify`/`JSON.parse` ourself results in an exception with a proper error message being thrown instead, so you know where the issue has occured.